### ISSUE TITLE
SIM300 (Yoda conditions) incorrectly triggered by CAPITAL variable

### ISSUE THREAD

**Issue by av223119** [astral-sh/ruff#10848]
The following code:

```python
if CLIENT_IPS != ["127.0.0.1"]:
    pass
```

triggers the error from ruff:
`SIM300 [*] Yoda conditions are discouraged, use `["127.0.0.1"] != CLIENT_IPS` instead`

First, as one can see this is not, in fact, a Yoda condition; but what is more interesting is that renaming the variable to lowercase eliminates the error! The following code does not trigger the SIM300:

```python
if client_ips != ["127.0.0.1"]:
    pass
```

ruff 0.3.5, SIM module enabled is everything needed to trigger the issue.

**Comment by charliermarsh** [comment]
Thankfully this is already fixed in preview. You can see here: https://play.ruff.rs/14c4fc15-f25d-450f-9bb3-516c8b07b86c. So running with `--preview` will resolve the issue. Otherwise, we'll likely stabilize this in Ruff v0.4.0.
